Source/Core/DolphinQt/Debugger/CodeDiffDialog.cpp

```diff
@@ -497,21 +620,25 @@ void CodeDiffDialog::OnSetBLR()
   if (!symbol)
     return;
 
-  {
-    auto& system = Core::System::GetInstance();
-    Core::CPUThreadGuard guard(system);
-    system.GetPowerPC().GetDebugInterface().SetPatch(guard, symbol->address, 0x4E800020);
-  }
+  MarkRowBLR(item->row());
+  if (m_autosave_check->isChecked())
+    SaveDataBackup();
 
-  int row = item->row();
+  auto& system = Core::System::GetInstance();
+  Core::CPUThreadGuard guard(system);
+  system.GetPowerPC().GetDebugInterface().SetPatch(guard, symbol->address, 0x4E800020);

```

Comment:
This still needs the scope per Jos's commit 7cecb28
{ } 
 > DolphinQt: Properly lock CPU before accessing emulated memory

> This fixes a problem I was having where using frame advance with the
debugger open would frequently cause panic alerts about invalid addresses
due to the CPU thread changing MSR.DR while the host thread was trying
to access memory.

Reply:
@JosJuice can you clarify if the braces around CPUThreadGuard are necessary or not? I just assumed it was necessary because thread related.

Reply:
Short answer: the braces should be there.

The purpose of CPUThreadGuard is to make sure the CPU thread isn't running while the guard exists, which lets the Host thread (which is basically the User Interface/Qt thread) do things like modify memory here without running into threading issues.

Because they prevent the CPU from running guards should exist for only as long as they're needed to avoid unnecessary performance penalties. The rest of the function doesn't need the guard, and the braces cause the guard to exist only within that scope, so afterward the CPU thread is free to run again instead of having to wait for the end of the OnSetBLR function.